bsdlib/README.rst

```diff
@@ -3,9 +3,131 @@
 BSD Library
 ###########
 
+The BSD library is Nordic Semiconductor's implementation of the BSD Socket API, which is a set of standard function calls that can be used in an application.
+This library aims to be compatible with the BSD socket standard to simplify writing and porting of existing applications. 
+
+The library has the following set of sockets:
+
+* 8 generic sockets that can be used generically between UDP, TCP, TLS, DTLS, and AT commands.
+* 1 GNSS/AGPS socket
+* 1 Modem DFU socket
+* 10 PDN sockets
+
+Overview
+********
+
+.. figure:: doc/images/bsd_lib_overview.svg
+   :alt: BSD library overview diagram
+
+   BSD library overview diagram
+
+The BSD library is the primary interface to operate the nRF9160 modem in order to establish LTE-M, NBIOT, and GNSS connections.
+The library abstracts away the details on memory management and synchronization leaving the application to focus on writing the application using standardized API calls.
+
+The current implementation of the BSD library supports the following protocols:
+
+* TCP (client/server)
+* UDP
+* RAW sockets
+* DTLS 1.2 client
+* TLS 1.2 client
+* AT commands
+* GNSS
+* AGPS
+* PDN management
+* Modem DFU
+
+.. note::
+   The DNS address management protocol is not yet implemented.
+
+The library implements the following standard BSD functions:
+   
+* socket
+* close
+* bind
+* listen
+* accept
+* connect
+* recvfrom
+* recv
+* read
+* sendto
+* send
+* write
+* getsockopt
+* setsockopt
+* fcntl
+* poll
+* inet_pton
+* inet_ntop
+* getaddrinfo
+* freeaddrinfo
+
+.. note::
+   The ``select`` function is not yet implemented.
+
+
+Library internals
+*****************
+
+The purpose of the BSD library is to communicate with the nRF9160 modem firmware.
+This is done by sending events and getting responses over the Inter Processor Communication peripheral (IPC).
+To facilitate data transfer between the application and the modem, payload data is set in the shared RAM.
+
+The modem contains a full IP and DTLS/TLS stack as well as GNSS.
+The library internally communicates over a Remote Procedure Call (RPC) protocol using the IPC and the shared RAM to utilize the modules inside the modem.
+The library provides address families, types, and protocols that dispatch the protocols to the right backend in the modem and thus communicates with this backend.
```

Comment:
Are there different backends in the modem? Is this relevant to the user?

Reply:
postponed

Reply:
I am not sure if this sentence make sense at all:

Maybe something like this:

By providing the right combinations of address families, types, and protocols to the library, will the library be able to dispatch data  to and from the different backends in the modem firmware.


that dispatch the protocols to the right backend in the modem and thus communicates with this backend.

Reply:
Changed to:
"You can dispatch data to and from the different modules in the modem firmware by providing the right combinations of address families, types, and protocols to the BSD library."

@evenl  please let me know if this is ok